core/src/test/scala/io/delta/tables/DeltaTableSuite.scala

```diff
@@ -168,6 +168,75 @@ class DeltaTableSuite extends QueryTest
     }
   }
 
+  test("isDeltaTableByPath - with _delta_log dir") {
+    withTempDir { dir =>
+      testData.write.format("delta").save(dir.getAbsolutePath)
+      assert(DeltaTable.isDeltaTableByPath(dir.getAbsolutePath))
+    }
+  }
+
+  test("isDeltaTableByPath - with empty _delta_log dir") {
+    withTempDir { dir =>
+      new File(dir, "_delta_log").mkdirs()
+      assert(!DeltaTable.isDeltaTableByPath(dir.getAbsolutePath))
+    }
+  }
+
+  test("isDeltaTableByPath - with no _delta_log dir") {
+    withTempDir { dir =>
+      assert(!DeltaTable.isDeltaTableByPath(dir.getAbsolutePath))
+    }
+  }
+
+  test("isDeltaTableByPath - with non-existent dir") {
+    withTempDir { dir =>
+      JavaUtils.deleteRecursively(dir)
+      assert(!DeltaTable.isDeltaTableByPath(dir.getAbsolutePath))
+    }
+  }
+
+  test("isDeltaTableByPath - with non-Delta table path") {
+    withTempDir { dir =>
+      testData.write.format("parquet").mode("overwrite").save(dir.getAbsolutePath)
+      assert(!DeltaTable.isDeltaTableByPath(dir.getAbsolutePath))
+    }
+  }
+
+  test("isDeltaTableByName - with table name") {
+    val tblName = "anotherDeltaTable"
+    withTable(tblName) {
+      testData.write.format("delta").saveAsTable(tblName)
+      assert(DeltaTable.isDeltaTableByName(tblName))
+    }
+  }
+
+  test("isDeltaTableByName - with fully qualified table name") {
+    withDatabase("delta") {
+        sql("CREATE DATABASE delta")
+        withTable("deltatablename") {
+          testData.write.format("delta").saveAsTable("delta.deltatablename")
+          assert(DeltaTable.isDeltaTableByName("delta.deltatablename"))
+      }
+    }
+  }
+
+  test("isDeltaTableByName - with view name") {
```

Comment:
can we add a positive case with view?

Reply:
Actually this negative test doesn't really test anything; all `withView` does is drop `viewName` after the test. Just change this to a positive test (you can kind of copy the test above for a temp view with forName)